<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>iSEEfier Workshop SCBC2024 • iSEEfierWorkshopSCBS2024</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="iSEEfier Workshop SCBC2024">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">iSEEfierWorkshopSCBS2024</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/workshop_iSEEfier.html">iSEEfier Workshop SCBC2024</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/NajlaAbassi/iSEEfierWorkshopSCBS2024" class="external-link">
    <span class="fa fa-github"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>iSEEfier Workshop SCBC2024</h1>
                        <h4 data-toc-skip class="author">Najla
Abassi</h4>
            <address class="author_afil">
      Institute of Medical Biostatistics, Epidemiology and Informatics
(IMBEI),
Mainz<br><a class="author_email" href="mailto:#"><p><a href="mailto:najla.abassi@uni-mainz.de" class="email">najla.abassi@uni-mainz.de</a>  Some materials in this
workshop were adapted from resources created by the <code>iSEE</code>
core development team</p></a>
      </address>
                  
            <h4 data-toc-skip class="date">14 November 2024</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/najlaabassi/iSEEfierWorkshopSCBS2024/blob/HEAD/vignettes/workshop_iSEEfier.Rmd" class="external-link"><code>vignettes/workshop_iSEEfier.Rmd</code></a></small>
      <div class="hidden name"><code>workshop_iSEEfier.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="streamlining-omics-data-visualization-with-isee-and-iseefier">Streamlining Omics Data Visualization with <code>iSEE</code> and
<code>iSEEfier</code><a class="anchor" aria-label="anchor" href="#streamlining-omics-data-visualization-with-isee-and-iseefier"></a>
</h2>
<div class="section level3">
<h3 id="introduction-to-isee">Introduction to <code>iSEE</code><a class="anchor" aria-label="anchor" href="#introduction-to-isee"></a>
</h3>
<p>The <em><a href="https://bioconductor.org/packages/3.20/iSEE" class="external-link">iSEE</a></em> package
is a flexible, powerful and extendible application to explore and
visualize any omics datas, including single cell and spatially resolved
data.</p>
<div class="section level4">
<h4 id="input-data">Input data<a class="anchor" aria-label="anchor" href="#input-data"></a>
</h4>
<p><em><a href="https://bioconductor.org/packages/3.20/iSEE" class="external-link">iSEE</a></em> was
designed around the <em><a href="https://bioconductor.org/packages/3.20/SummarizedExperiment" class="external-link">SummarizedExperiment</a></em>
class, a container widely used throughout the <em>Bioconductor</em>
project. Briefly, the <em><a href="https://bioconductor.org/packages/3.20/SummarizedExperiment" class="external-link">SummarizedExperiment</a></em>
class provides a container keeping matrices of assay data, sample
metadata, and feature metadata.</p>
<p><img src="img%2FSummarizedExperiment.png" alt="SummarizedExperiment (source: the package vignette; https://bioconductor.org/packages/SummarizedExperiment/)" width="100%"></p>
<p>In this workshop we will use a demonstration dataset of single cell
RNAseq from Tasic et al. (2016) “Adult mouse cortical cell taxonomy
revealed by single cell transcriptomics” <a href="https://doi.org/10.1038/nn.4216" class="external-link">doi:10.1038/nn.4216</a><span class="citation">(Tasic et al. 2016)</span>.<br>
The data is made available via the <em><a href="https://bioconductor.org/packages/3.20/scRNAseq" class="external-link">scRNAseq</a></em>
Bioconductor package, and is stored in an object of class <em><a href="https://bioconductor.org/packages/3.20/SingleCellExperiment" class="external-link">SingleCellExperiment</a></em>,
which is a class that inherits from the
<code>SummarizedExperiment</code> class. It includes additional slots
and methods like metadata related to individual cells and their
associated dimensionality reductions.</p>
<p><img src="img%2FSingleCellExperiment.png" alt="SingleCellExperiment (source: the package vignette; \&lt;https://bioconductor.org/packages/SingleCellExperiment/)" width="100%"></p>
<p>We processed the data beforehand to save some time.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://najlaabassi.github.io/iSEEfierWorkshopSCBS2024/">"iSEEfierWorkshopSCBS2024"</a></span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu">iSEEfierWorkshopSCBS2024</span><span class="fu">::</span><span class="va"><a href="../reference/sce_allen.html">sce_allen</a></span></span>
<span><span class="va">sce</span></span>
<span><span class="co">#&gt; class: SingleCellExperiment </span></span>
<span><span class="co">#&gt; dim: 20816 379 </span></span>
<span><span class="co">#&gt; metadata(2): SuppInfo which_qc</span></span>
<span><span class="co">#&gt; assays(2): counts logcounts</span></span>
<span><span class="co">#&gt; rownames(20816): 0610007P14Rik 0610009B22Rik ... Zzef1 Zzz3</span></span>
<span><span class="co">#&gt; rowData names(0):</span></span>
<span><span class="co">#&gt; colnames(379): SRR2140028 SRR2140022 ... SRR2139341 SRR2139336</span></span>
<span><span class="co">#&gt; colData names(23): NREADS NALIGNED ... passes_qc_checks_s sizeFactor</span></span>
<span><span class="co">#&gt; reducedDimNames(3): PCA TSNE UMAP</span></span>
<span><span class="co">#&gt; mainExpName: endogenous</span></span>
<span><span class="co">#&gt; altExpNames(1): ERCC</span></span></code></pre></div>
<p>Looking at the object displayed above, we can tell that it
contains:</p>
<ul>
<li>Two <code>assays</code>. The first one called <code>counts</code>,
and corresponds to a matrix measuring 20,816 genes in 379 cells. And The
second assay is <code>logcounts</code>, which is the log transformed
counts.</li>
<li>Twenty-three columns of cell metadata (<code>colData</code>).</li>
<li>No gene metadata (<code>rowData</code>), but only the basic
information in the <code>rownames</code>.</li>
<li>Three reduced dimensions : <code>PCA</code>, <code>TSNE</code>, and
<code>UMAP</code>.</li>
</ul>
<p>Also of note:</p>
<ul>
<li>Gene symbols are in use for the row names.</li>
<li>Arbitrary cell names are in use for column names (related to the SRA
run information).</li>
<li>Some object-level metadata is present.</li>
<li>An alternative experiment called <code>ERCC</code> is present (we
will not use it) - this is related to the <a href="https://www.thermofisher.com/order/catalog/product/de/en/4456740" class="external-link">ERCC
Spike-in mix</a>.</li>
</ul>
</div>
<div class="section level4">
<h4 id="the-default-isee-app">The default <code>iSEE</code> app<a class="anchor" aria-label="anchor" href="#the-default-isee-app"></a>
</h4>
<p>Now we can launch an <em><a href="https://bioconductor.org/packages/3.20/iSEE" class="external-link">iSEE</a></em>
instance and start exploring this dataset using the <code><a href="https://rdrr.io/pkg/iSEE/man/iSEE.html" class="external-link">iSEE()</a></code>
function without any further argument. This will produce an app using
the default configuration; that is, the app instance will include one
panel of each built-in class for which the relevant information is
available in the <code>SingleCellExperiment</code> object.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">app</span> <span class="op">&lt;-</span> <span class="fu">iSEE</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/iSEE/man/iSEE.html" class="external-link">iSEE</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">shiny</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/shiny/man/runApp.html" class="external-link">runApp</a></span><span class="op">(</span><span class="va">app</span>,launch.browser <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="img%2FiSEE_default.png" width="100%"></p>
</div>
<div class="section level4">
<h4 id="the-built-in-panel-types">The built-in panel types<a class="anchor" aria-label="anchor" href="#the-built-in-panel-types"></a>
</h4>
<div class="section level5">
<h5 id="reduceddimensionplot">
<code>ReducedDimensionPlot</code><a class="anchor" aria-label="anchor" href="#reduceddimensionplot"></a>
</h5>
<p>The reduced dimension plot can display any reduced dimension
representation that is present in the <code>reducedDim</code> slot of
the <code>SingleCellExperiment</code> object.</p>
<p>Note that this slot is not defined for the base
<code>SummarizedExperiment</code> class, in which case the user
interface does not allow the inclusion of panels of this type.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">shiny</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/shiny/man/runApp.html" class="external-link">runApp</a></span><span class="op">(</span><span class="va">app</span>,launch.browser <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="img%2FiSEE_RedDimPlot.png" width="70%" style="display: block; margin: auto;"></p>
</div>
<div class="section level5">
<h5 id="columndataplot">
<code>ColumnDataPlot</code><a class="anchor" aria-label="anchor" href="#columndataplot"></a>
</h5>
<p>The column data plot can display one or two of the provided column
annotations (from the <code>colData</code> slot). Depending on the class
of the selected annotations, the panel shows either a scatter plot, a
violin plot, or a Hinton diagram <span class="citation">(Hinton and
Shallice 1991; Bremner, Gotts, and Denham 1994)</span>.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">shiny</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/shiny/man/runApp.html" class="external-link">runApp</a></span><span class="op">(</span><span class="va">app</span>,launch.browser <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="img%2FiSEE_ColDataPlot.png" width="70%" style="display: block; margin: auto;"></p>
</div>
<div class="section level5">
<h5 id="complexheatmapplot">
<code>ComplexHeatmapPlot</code><a class="anchor" aria-label="anchor" href="#complexheatmapplot"></a>
</h5>
<p>The complex heatmap panel displays, for any assay, the observed
values for a subset of the features across the samples.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">shiny</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/shiny/man/runApp.html" class="external-link">runApp</a></span><span class="op">(</span><span class="va">app</span>,launch.browser <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="img%2FiSEE_Heatmap.png" width="70%" style="display: block; margin: auto;"></p>
</div>
<div class="section level5">
<h5 id="featureassayplot">
<code>FeatureAssayPlot</code><a class="anchor" aria-label="anchor" href="#featureassayplot"></a>
</h5>
<p>The feature assay plot displays the observed values for one feature
across the samples. It is also possible to plot the observed values for
two features, in a scatter plot.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">shiny</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/shiny/man/runApp.html" class="external-link">runApp</a></span><span class="op">(</span><span class="va">app</span>,launch.browser <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="img%2FiSEE_FeatureAssayPlt.png" width="70%" style="display: block; margin: auto;"></p>
</div>
<div class="section level5">
<h5 id="sampleassayplot">
<code>SampleAssayPlot</code><a class="anchor" aria-label="anchor" href="#sampleassayplot"></a>
</h5>
<p>Analogous to the <a href="#featassayplot">Feature assay plot</a>
above, the Sample assay plot shows the observed values for all features,
for one of the samples. It is also possible to plot the observed values
for two samples, in a scatter plot.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">shiny</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/shiny/man/runApp.html" class="external-link">runApp</a></span><span class="op">(</span><span class="va">app</span>,launch.browser <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="img%2FiSEE_SampleAssayPlt.png" width="70%" style="display: block; margin: auto;"></p>
</div>
<div class="section level5">
<h5 id="rowdatatable">
<code>RowDataTable</code><a class="anchor" aria-label="anchor" href="#rowdatatable"></a>
</h5>
<p>The row data table displays all information provided in the
<code>rowData</code> slot of the <code>SummarizedExperiment</code>
object, leveraging the interactivity provided by the <em><a href="https://CRAN.R-project.org/package=DT" class="external-link">DT</a></em> package.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">shiny</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/shiny/man/runApp.html" class="external-link">runApp</a></span><span class="op">(</span><span class="va">app</span>,launch.browser <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="img%2FiSEE_RowDataTable.png" width="70%" style="display: block; margin: auto;"></p>
</div>
<div class="section level5">
<h5 id="columndatatable">
<code>ColumnDataTable</code><a class="anchor" aria-label="anchor" href="#columndatatable"></a>
</h5>
<p>Analogous to the <a href="#rowdatatable">Row data table</a> above,
the column data table displays all information provided in the
<code>colData</code> slot of the <code>SummarizedExperiment</code>
object.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">shiny</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/shiny/man/runApp.html" class="external-link">runApp</a></span><span class="op">(</span><span class="va">app</span>,launch.browser <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="img%2FiSEE_ColDataTable.png" width="70%" style="display: block; margin: auto;"></p>
</div>
</div>
<div class="section level4">
<h4 id="collapsible-boxes-with-display-controls">Collapsible boxes with display controls<a class="anchor" aria-label="anchor" href="#collapsible-boxes-with-display-controls"></a>
</h4>
<div class="section level5">
<h5 id="data-parameters">Data parameters<a class="anchor" aria-label="anchor" href="#data-parameters"></a>
</h5>
<p>Each plot panel type has a <code>Data parameters</code> collapsible
box. This box has different content for each panel type, but in all
cases it lets the user control the data that is displayed in the
plot.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">shiny</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/shiny/man/runApp.html" class="external-link">runApp</a></span><span class="op">(</span><span class="va">app</span>,launch.browser <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="img%2FiSEE_data_param.png" width="70%" style="display: block; margin: auto;"></p>
</div>
<div class="section level5">
<h5 id="visual-parameters">Visual parameters<a class="anchor" aria-label="anchor" href="#visual-parameters"></a>
</h5>
<p>In contrast to the <code>Data parameters</code> collapsible box that
lets users control <em>what</em> is displayed in the plot, the
<code>Visual parameters</code> box lets users control <em>how</em> the
information is displayed.</p>
<p>This collapsible box contains the controls to change the size, shape,
opacity, and color of the points, to facet the plot by any available
categorical annotation, to subsample points for increased speed of plot
rendering, and to control how legends are displayed.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">shiny</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/shiny/man/runApp.html" class="external-link">runApp</a></span><span class="op">(</span><span class="va">app</span>,launch.browser <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="img%2FiSEE_viz_param.png" width="70%" style="display: block; margin: auto;"></p>
</div>
<div class="section level5">
<h5 id="selection-parameters">Selection parameters<a class="anchor" aria-label="anchor" href="#selection-parameters"></a>
</h5>
<p>The <code>Selection parameters</code> collapsible box provides
controls to transfer selections of points (features or samples) between
panels.</p>
<p>We demonstrate examples of point transmission in the separate
vignette of <a href="recipes.html">workshop recipes</a></p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">shiny</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/shiny/man/runApp.html" class="external-link">runApp</a></span><span class="op">(</span><span class="va">app</span>,launch.browser <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p><img src="img%2FiSEE_selec_param.png" width="70%" style="display: block; margin: auto;"></p>
</div>
</div>
</div>
<div class="section level3">
<h3 id="introduction-to-iseefier">Introduction to <code>iSEEfier</code><a class="anchor" aria-label="anchor" href="#introduction-to-iseefier"></a>
</h3>
<p>Now we have seen how powerful of a tool <code>iSEE</code> can be. As
we previously learned, many tasks can be accomplished simply by running
the command:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/iSEE/man/iSEE.html" class="external-link">iSEE</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span></span></code></pre></div>
<p>However, customizing our visualization session can be cumbersome, as
it may require adding or removing different panels based on our needs.
This process often includes multiple steps, such as selecting a gene of
interest or coloring by a specific <code>colData</code>, and may even
involve writing multiple lines of code.</p>
<p>To make this process simpler, we can use the <em><a href="https://bioconductor.org/packages/3.20/iSEEfier" class="external-link">iSEEfier</a></em>
package, which streamlines the setup (or if you will, firing up) of an
<code>iSEE</code> instance with just a small chunk of code, avoiding the
need to configure each <code>iSEE</code> panel individually.</p>
<p>In this section, we will illustrate a simple example of how to use
<em><a href="https://bioconductor.org/packages/3.20/iSEEfier" class="external-link">iSEEfier</a></em>.
We will use the same allen data we worked with during this workshop.</p>
<div class="section level5">
<h5 id="iseeinit">
<code>iSEEinit</code><a class="anchor" aria-label="anchor" href="#iseeinit"></a>
</h5>
<p>For example, we can be interested in visualizing the expression of
<em>Bcl6</em>, <em>Arf5</em>, <em>Cxcr5</em> and <em>Grip1</em> genes
all at once. We start by providing a couple of parameters:</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Define the list of genes</span></span>
<span><span class="va">feature_list_1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Bcl6"</span>, <span class="st">"Arf5"</span>, <span class="st">"Cxcr5"</span>,<span class="st">"Grip1"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Define the cluster/cell type </span></span>
<span><span class="va">cluster_1</span> <span class="op">&lt;-</span> <span class="st">"Primary.Type"</span></span></code></pre></div>
<p>It is also possible to load the feature list as a
<code>data.frame</code>, if our genes of interest already existed in a
table. Now we can pass these parameters into <code><a href="https://rdrr.io/pkg/iSEEfier/man/iSEEinit.html" class="external-link">iSEEinit()</a></code> to
create a customized initial configuration:</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Create an initial state with iSEEinit</span></span>
<span><span class="va">initial_1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/iSEEfier/man/iSEEinit.html" class="external-link">iSEEinit</a></span><span class="op">(</span><span class="va">sce</span>,</span>
<span>                      features <span class="op">=</span> <span class="va">feature_list_1</span>,</span>
<span>                      clusters <span class="op">=</span> <span class="va">cluster_1</span>,</span>
<span>                      add_markdown_panel <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p>The rest can be as easy as passing this initial to the
<code><a href="https://rdrr.io/pkg/iSEE/man/iSEE.html" class="external-link">iSEE()</a></code> call:</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">shiny</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/shiny/man/runApp.html" class="external-link">runApp</a></span><span class="op">(</span><span class="va">app</span>,launch.browser <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p>This is how it would look like:</p>
<p><img src="img%2Fisee-on-fier.png" width="100%"></p>
<p>While we are visualizing the expression of these genes, we might want
to take some notes (gene X is more expressed in a certain cell
type/cluster than some others, maybe we are trying to annotate the cells
ourselves if the annotation wasn’t available…).<br>
For this, we used the argument <code>add_markdown_panel = TRUE</code>.
It will display a <code>MarkdownBoard</code> panel where we can note our
observations without leaving the app.</p>
</div>
<div class="section level5">
<h5 id="iseemarker">
<code>iSEEmarker</code><a class="anchor" aria-label="anchor" href="#iseemarker"></a>
</h5>
<p>In many cases, we are interested in determining the identity of our
cell clusters, or further subset our cells types. That’s where
<code><a href="https://rdrr.io/pkg/iSEEfier/man/iSEEmarker.html" class="external-link">iSEEmarker()</a></code> comes in handy. Similar to
<code><a href="https://rdrr.io/pkg/iSEEfier/man/iSEEinit.html" class="external-link">iSEEinit()</a></code>, we need a couple of parameters to be set:</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Define the cluster/cell type </span></span>
<span><span class="va">cluster_1</span> <span class="op">&lt;-</span> <span class="st">"Primary.Type"</span></span>
<span></span>
<span><span class="co">## Define the groups/conditions</span></span>
<span><span class="va">group_1</span> <span class="op">&lt;-</span> <span class="st">"driver_1_s"</span></span></code></pre></div>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">initial_2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/iSEEfier/man/iSEEmarker.html" class="external-link">iSEEmarker</a></span><span class="op">(</span></span>
<span>  sce <span class="op">=</span> <span class="va">sce</span>,</span>
<span>  clusters <span class="op">=</span> <span class="va">cluster_1</span>,</span>
<span>  groups <span class="op">=</span> <span class="va">group_1</span>,</span>
<span>  selection_plot_format <span class="op">=</span> <span class="st">"ColumnDataPlot"</span><span class="op">)</span></span></code></pre></div>
<p>This function returns a list of panels, with the goal of visualizing
the expression of marker genes selected from the
<code>DynamicMarkerTable</code> in each cell cell type. Unlike
<code><a href="https://rdrr.io/pkg/iSEEfier/man/iSEEinit.html" class="external-link">iSEEinit()</a></code>, which requires us to specify a list of genes,
<code><a href="https://rdrr.io/pkg/iSEEfier/man/iSEEmarker.html" class="external-link">iSEEmarker()</a></code> utilizes the <code>DynamicMarkerTable</code>,
a custom panel in <code>iSEE</code>, that performs statistical testing
through the <code>findMarkers()</code> function from the <em><a href="https://bioconductor.org/packages/3.20/scran" class="external-link">scran</a></em>
package. To start exploring the marker genes of each cell type with
<code>iSEE</code>, we run as usual:</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">shiny</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/shiny/man/runApp.html" class="external-link">runApp</a></span><span class="op">(</span><span class="va">app</span>,launch.browser <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p>This is how it would look like:</p>
<p><img src="img%2FiSEEmarker.png" width="100%"></p>
</div>
<div class="section level5">
<h5 id="iseenrich">
<code>iSEEnrich</code><a class="anchor" aria-label="anchor" href="#iseenrich"></a>
</h5>
<p>At times, it can be useful to explore particular sets of features and
their associated genes. This is where <code>iSEEnrich</code> proves to
be especially valuable. We will first set up the necessary
arguments:</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Set which terms to use</span></span>
<span><span class="va">GO_collection</span> <span class="op">&lt;-</span> <span class="st">"GO"</span></span>
<span></span>
<span><span class="co">## Set the organism via the orgDb package name</span></span>
<span><span class="va">Mm_organism</span> <span class="op">&lt;-</span> <span class="st">"org.Mm.eg.db"</span></span>
<span></span>
<span><span class="co">## Set the gene identifier type</span></span>
<span><span class="va">gene_id</span> <span class="op">&lt;-</span> <span class="st">"SYMBOL"</span></span>
<span></span>
<span><span class="co">## Define the cluster info</span></span>
<span><span class="va">cluster_1</span> <span class="op">&lt;-</span> <span class="st">"Primary.Type"</span></span>
<span><span class="co">## Define the group info</span></span>
<span><span class="va">group_1</span> <span class="op">&lt;-</span> <span class="st">"driver_1_s"</span></span>
<span><span class="co">## Define the plot type for the reduced dimension</span></span>
<span><span class="va">reddim_type</span> <span class="op">&lt;-</span> <span class="st">"TSNE"</span></span></code></pre></div>
<p>Now let’s create this initial setup for <code>iSEE</code> using
<code><a href="https://rdrr.io/pkg/iSEEfier/man/iSEEnrich.html" class="external-link">iSEEnrich()</a></code></p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">results</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/iSEEfier/man/iSEEnrich.html" class="external-link">iSEEnrich</a></span><span class="op">(</span></span>
<span>  sce <span class="op">=</span> <span class="va">sce</span>,</span>
<span>  collection <span class="op">=</span> <span class="va">GO_collection</span>,</span>
<span>  gene_identifier <span class="op">=</span> <span class="va">gene_id</span>,</span>
<span>  organism <span class="op">=</span> <span class="va">Mm_organism</span>,</span>
<span>  clusters <span class="op">=</span> <span class="va">cluster_1</span>,</span>
<span>  reddim_type <span class="op">=</span> <span class="va">reddim_type</span>,</span>
<span>  groups <span class="op">=</span> <span class="va">group_1</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span></code></pre></div>
<p><code>iSEEnrich</code> will specifically return a list with the
updated <code>sce</code> object and its associated <code>initial</code>
configuration. To start the <code>iSEE</code> instance we run:</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">shiny</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/shiny/man/runApp.html" class="external-link">runApp</a></span><span class="op">(</span><span class="va">app</span>,launch.browser <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p>This is how it would look like:</p>
<p><img src="img%2FiSEEnrich.png" width="100%"></p>
</div>
<div class="section level5">
<h5 id="view_initial_">
<code>view_initial_*</code><a class="anchor" aria-label="anchor" href="#view_initial_"></a>
</h5>
<p>We can check the initial’s content, or how the included panels are
linked between each other without running the app with
<code><a href="https://rdrr.io/pkg/iSEEfier/man/view_initial_tiles.html" class="external-link">view_initial_tiles()</a></code> and
<code><a href="https://rdrr.io/pkg/iSEEfier/man/view_initial_network.html" class="external-link">view_initial_network()</a></code>:</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Display a graphical representation of the initial configuration, where the panels are identified by their corresponding colors</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/iSEEfier/man/view_initial_tiles.html" class="external-link">view_initial_tiles</a></span><span class="op">(</span>initial <span class="op">=</span> <span class="va">initial_1</span><span class="op">)</span></span></code></pre></div>
<p><img src="workshop_iSEEfier_files/figure-html/unnamed-chunk-34-1.png" width="672"></p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Display a network visualization for the panels</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/iSEEfier/man/view_initial_network.html" class="external-link">view_initial_network</a></span><span class="op">(</span><span class="va">initial_1</span>, plot_format <span class="op">=</span> <span class="st">"igraph"</span><span class="op">)</span></span></code></pre></div>
<p><img src="workshop_iSEEfier_files/figure-html/unnamed-chunk-35-1.png" width="768"></p>
<pre><code><span><span class="co">#&gt; IGRAPH 3dbf241 DN-- 17 5 -- </span></span>
<span><span class="co">#&gt; + attr: name (v/c), color (v/c)</span></span>
<span><span class="co">#&gt; + edges from 3dbf241 (vertex names):</span></span>
<span><span class="co">#&gt; [1] ReducedDimensionPlot1-&gt;ColumnDataPlot1  </span></span>
<span><span class="co">#&gt; [2] ReducedDimensionPlot2-&gt;ColumnDataPlot1  </span></span>
<span><span class="co">#&gt; [3] ReducedDimensionPlot3-&gt;ColumnDataPlot1  </span></span>
<span><span class="co">#&gt; [4] ReducedDimensionPlot4-&gt;ColumnDataPlot1  </span></span>
<span><span class="co">#&gt; [5] ReducedDimensionPlot5-&gt;FeatureAssayPlot5</span></span></code></pre>
<p>Another alternative for network visualization would use the
interactive widget provided by <code>visNetwork</code>:</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/iSEEfier/man/view_initial_network.html" class="external-link">view_initial_network</a></span><span class="op">(</span><span class="va">initial_1</span>, plot_format <span class="op">=</span> <span class="st">"visNetwork"</span><span class="op">)</span></span></code></pre></div>
<p>It is also possible to combine multiple initials into one:</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">merged_config</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/iSEEfier/man/glue_initials.html" class="external-link">glue_initials</a></span><span class="op">(</span><span class="va">initial_1</span>, <span class="va">initial_2</span><span class="op">)</span></span>
<span><span class="co">## Check out the content of merged_config</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/iSEEfier/man/view_initial_tiles.html" class="external-link">view_initial_tiles</a></span><span class="op">(</span>initial <span class="op">=</span> <span class="va">merged_config</span><span class="op">)</span></span></code></pre></div>
<p><code><a href="https://rdrr.io/pkg/iSEEfier/man/iSEEfier-pkg.html" class="external-link">?iSEEfier</a></code> is always your friend whenever you need
further documentation on the package/a certain function and how to use
it.</p>
<p>Feel free to explore previous <code>iSEE</code> demos and
<code>workshops</code>, as well as the <code>iSEEfier</code> vignette,
for a deeper dive into using these tools. They’ll help you make the most
of them :)<br>
If you look for a place to start and have a full one-day workshop,
please refer to <a href="https://isee.github.io/iUSEiSEE/" class="external-link"><code>https://isee.github.io/iUSEiSEE/</code></a>.</p>
</div>
</div>
<div class="section level3">
<h3 class="unnumbered" id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h3>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.4.1 (2024-06-14)</span></span>
<span><span class="co">#&gt; Platform: aarch64-apple-darwin20</span></span>
<span><span class="co">#&gt; Running under: macOS Ventura 13.6</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRblas.0.dylib </span></span>
<span><span class="co">#&gt; LAPACK: /Library/Frameworks/R.framework/Versions/4.4-arm64/Resources/lib/libRlapack.dylib;  LAPACK version 3.12.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt; [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: Europe/Berlin</span></span>
<span><span class="co">#&gt; tzcode source: internal</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">#&gt; [8] base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] iSEEfierWorkshopSCBS2024_1.0.0 iSEEfier_1.1.2                </span></span>
<span><span class="co">#&gt;  [3] iSEE_2.17.4                    SingleCellExperiment_1.27.2   </span></span>
<span><span class="co">#&gt;  [5] SummarizedExperiment_1.35.1    Biobase_2.65.1                </span></span>
<span><span class="co">#&gt;  [7] GenomicRanges_1.57.1           GenomeInfoDb_1.41.1           </span></span>
<span><span class="co">#&gt;  [9] IRanges_2.39.2                 S4Vectors_0.43.2              </span></span>
<span><span class="co">#&gt; [11] BiocGenerics_0.51.1            MatrixGenerics_1.17.0         </span></span>
<span><span class="co">#&gt; [13] matrixStats_1.4.1             </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] DBI_1.2.3               rlang_1.1.4             magrittr_2.0.3         </span></span>
<span><span class="co">#&gt;   [4] shinydashboard_0.7.2    clue_0.3-65             GetoptLong_1.0.5       </span></span>
<span><span class="co">#&gt;   [7] RSQLite_2.3.7           compiler_4.4.1          mgcv_1.9-1             </span></span>
<span><span class="co">#&gt;  [10] png_0.1-8               systemfonts_1.1.0       vctrs_0.6.5            </span></span>
<span><span class="co">#&gt;  [13] iSEEhex_1.7.0           pkgconfig_2.0.3         shape_1.4.6.1          </span></span>
<span><span class="co">#&gt;  [16] crayon_1.5.3            fastmap_1.2.0           XVector_0.45.0         </span></span>
<span><span class="co">#&gt;  [19] fontawesome_0.5.2       utf8_1.2.4              promises_1.3.0         </span></span>
<span><span class="co">#&gt;  [22] rmarkdown_2.28          UCSC.utils_1.1.0        shinyAce_0.4.2         </span></span>
<span><span class="co">#&gt;  [25] ragg_1.3.3              bit_4.5.0               xfun_0.47              </span></span>
<span><span class="co">#&gt;  [28] zlibbioc_1.51.1         cachem_1.1.0            jsonlite_1.8.9         </span></span>
<span><span class="co">#&gt;  [31] blob_1.2.4              listviewer_4.0.0        highr_0.11             </span></span>
<span><span class="co">#&gt;  [34] later_1.3.2             DelayedArray_0.31.11    parallel_4.4.1         </span></span>
<span><span class="co">#&gt;  [37] cluster_2.1.6           R6_2.5.1                bslib_0.8.0            </span></span>
<span><span class="co">#&gt;  [40] RColorBrewer_1.1-3      jquerylib_0.1.4         Rcpp_1.0.13            </span></span>
<span><span class="co">#&gt;  [43] iterators_1.0.14        knitr_1.48              org.Mm.eg.db_3.19.1    </span></span>
<span><span class="co">#&gt;  [46] BiocBaseUtils_1.7.3     httpuv_1.6.15           Matrix_1.7-0           </span></span>
<span><span class="co">#&gt;  [49] splines_4.4.1           igraph_2.0.3            tidyselect_1.2.1       </span></span>
<span><span class="co">#&gt;  [52] rstudioapi_0.16.0       abind_1.4-8             yaml_2.3.10            </span></span>
<span><span class="co">#&gt;  [55] iSEEu_1.17.0            doParallel_1.0.17       codetools_0.2-20       </span></span>
<span><span class="co">#&gt;  [58] miniUI_0.1.1.1          lattice_0.22-6          tibble_3.2.1           </span></span>
<span><span class="co">#&gt;  [61] withr_3.0.1             KEGGREST_1.45.1         shiny_1.9.1            </span></span>
<span><span class="co">#&gt;  [64] evaluate_1.0.0          desc_1.4.3              Biostrings_2.73.1      </span></span>
<span><span class="co">#&gt;  [67] circlize_0.4.16         pillar_1.9.0            BiocManager_1.30.25    </span></span>
<span><span class="co">#&gt;  [70] DT_0.33                 foreach_1.5.2           shinyjs_2.1.0          </span></span>
<span><span class="co">#&gt;  [73] generics_0.1.3          ggplot2_3.5.1           munsell_0.5.1          </span></span>
<span><span class="co">#&gt;  [76] scales_1.3.0            BiocStyle_2.33.1        xtable_1.8-4           </span></span>
<span><span class="co">#&gt;  [79] glue_1.7.0              tools_4.4.1             hexbin_1.28.4          </span></span>
<span><span class="co">#&gt;  [82] colourpicker_1.3.0      visNetwork_2.1.2        fs_1.6.4               </span></span>
<span><span class="co">#&gt;  [85] grid_4.4.1              AnnotationDbi_1.67.0    colorspace_2.1-1       </span></span>
<span><span class="co">#&gt;  [88] nlme_3.1-166            GenomeInfoDbData_1.2.12 vipor_0.4.7            </span></span>
<span><span class="co">#&gt;  [91] cli_3.6.3               textshaping_0.4.0       fansi_1.0.6            </span></span>
<span><span class="co">#&gt;  [94] viridisLite_0.4.2       S4Arrays_1.5.7          ComplexHeatmap_2.21.0  </span></span>
<span><span class="co">#&gt;  [97] dplyr_1.1.4             gtable_0.3.5            rintrojs_0.3.4         </span></span>
<span><span class="co">#&gt; [100] sass_0.4.9              digest_0.6.37           SparseArray_1.5.36     </span></span>
<span><span class="co">#&gt; [103] ggrepel_0.9.6           farver_2.1.2            rjson_0.2.23           </span></span>
<span><span class="co">#&gt; [106] htmlwidgets_1.6.4       memoise_2.0.1           htmltools_0.5.8.1      </span></span>
<span><span class="co">#&gt; [109] pkgdown_2.1.1           lifecycle_1.0.4         httr_1.4.7             </span></span>
<span><span class="co">#&gt; [112] shinyWidgets_0.8.7      GlobalOptions_0.1.2     mime_0.12              </span></span>
<span><span class="co">#&gt; [115] bit64_4.5.2</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h3>
<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-Bremner1994-hintonplots" class="csl-entry">
Bremner, Frederick J, Stephen J Gotts, and Dina L Denham. 1994.
<span>“Hinton Diagrams: Viewing Connection Strengths in Neural
Networks.”</span> <em>Behav. Res. Methods Instrum. Comput.</em> 26 (2):
215–18.
</div>
<div id="ref-Hinton1991-hintondiagram" class="csl-entry">
Hinton, Geoffrey E, and Tim Shallice. 1991. <span>“Lesioning an
Attractor Network: Investigations of Acquired Dyslexia.”</span>
<em>Psychological Review</em> 98 (1): 74–95.
</div>
<div id="ref-Tasic2016-ga" class="csl-entry">
Tasic, Bosiljka, Vilas Menon, Thuc Nghi Nguyen, Tae Kyung Kim, Tim
Jarsky, Zizhen Yao, Boaz Levi, et al. 2016. <span>“Adult Mouse Cortical
Cell Taxonomy Revealed by Single Cell Transcriptomics.”</span> <em>Nat.
Neurosci.</em> 19 (2): 335–46.
</div>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Najla Abassi.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer>
</div>






  </body>
</html>
